### DISCUSSION TITLE
How get block side which one is looking player

### DISCUSSION THREAD

**Discussion by Intezium** [FabricMC/fabric-api#2449] [Mod Dev Support] [answered]
The image posted is the general idea of what I am trying to say..
![image](https://user-images.githubusercontent.com/20191322/183257602-3f650154-3cde-48f1-98da-18b43b60f3b6.png)

**Comment by warjort** [comment, marked as answer]
It depends where you are in the code and what kind of raycast/HitResult you find acceptable.

If your HitResult is hitting a block you will have a BlockHitResult which has a getSide().

Assuming where you are in the code doesn't have a HitResult parameter:

On the client there is ````MinecraftClient.crosshairTarget```` which may or may not be a ````BlockHitResult````
See MinecraftClient.doItemUse() for example.

You can do the calculation from scratch using something like:
````
		HitResult hit = entity.raycast(20, 0, false); // 20 is distance used by the DebugHud for "looking at block", false means ignore fluids
		if (hit.getType() == HitResult.Type.BLOCK) {
			BlockHitResult blockHit = (BlockHitResult) hit;
			if (blockHit.getSide() == whatever) {
				// do stuff
			}
		}
````

On the client the entity would be MinecraftClient.cameraEntity or on the server you would use the ServerPlayerEntity.

**Comment by SplashAni** [comment]
You can simply use 
`            BlockHitResult bhr = new BlockHitResult(Vec3d.of(pos),mc.player.getHorizontalFacing(),pos,false);
`
